src/Digdir.Tool.Dialogporten.SlackNotifier/ConvertAlertToSlackMessage.cs

```diff
@@ -0,0 +1,85 @@
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using System.Linq;
+using System.Net.Http;
+using System.Data;
+using Azure.Identity;
+using Azure.Core;
+using System.Net.Http.Headers;
+using System.Net.Http.Json;
+using System.Collections.Generic;
+using System;
+
+namespace Digdir.Tool.Dialogporten.SlackNotifier;
+
+public static class ConvertAlertToSlackMessage
+{
+    private static readonly DefaultAzureCredential _credentials = new();
+    private static readonly HttpClient _httpClient = new();
+    private const string encodedKqlQuery = "H4sIAAAAAAAAA0utSE4tKMnMzyvmqlHIL0pJLVJIqlQoycxNLS5JzC1QSEktTgYAbgDhFSQAAAA%253D/timespan/P1D";
+
+    [FunctionName("ConvertAlertToSlackMessage")]
+    public static async Task<IActionResult> Run(
+        [HttpTrigger(AuthorizationLevel.Function, "post", Route = null)] HttpRequest req,
+        ILogger log)
+    {
+        log.LogInformation("C# HTTP trigger function processed a request.");
+        using var streamReader = new StreamReader(req.Body);
+        var requestBody = await streamReader.ReadToEndAsync();
+        var azureAlertRequest = JsonConvert.DeserializeObject<AzureAlert>(requestBody);
+        var responses = await QueryAppInsights(azureAlertRequest);
+
+        var link = azureAlertRequest.Data.AlertContext.Condition.AllOf
+            .Select(x => x.LinkToFilteredSearchResultsUI)
+            .FirstOrDefault(x => !string.IsNullOrWhiteSpace(x));
+        link = RemoveQuery(link) + encodedKqlQuery;
+
+        var asciiTables = responses.SelectMany(x => x.Tables)
+            .Select(table => Enumerable.Empty<List<object>>()
+                .Append(table.Columns.Select(x => (object)x.Name))
+                .Concat(table.Rows)
+                .ToAsciiTable());
+
+        await _httpClient.PostAsJsonAsync(Environment.GetEnvironmentVariable("SlackWebhookUrl"), new
```

Comment:
Vurder å ha med en local.settings.json.COPYME elns i repoet som en kan kopiere til local.settings.json og som inneholder en (placeholder) verdi for denne miljøvariablen

Reply:
Med IOC fikk jeg brukt gode gamle options pattern. Nå er settingene i SlackOptions.cs, som injectes på rett plass. Har også lagt med en copyme.